opty/tests/test_direct_collocation.py

```diff
@@ -1907,3 +1907,161 @@ def test_attributes_read_only():
+        grad = np.zeros_like(free)
+        l1 = 2*num_nodes
+        l2 = 3*num_nodes
+        grad[l1: l2] = 2.0*free[l1:l2]*interval_value
+        return grad
+
+    instance_constraints = (
+        x.func(t0),
+        ux.func(t0),
+        x.func(tf) - 1.0,
+        ux.func(tf),
+    )
+
+    bounds= {
+        a1: (-1.0, 1.0),
+        a2: (-1.0, 1.0),
+        a3: (-1.0, 1.0),
+
+        u1: (-1.0, 1.0),
+        u2: (-1.0, 1.0),
+        u3: (-1.0, 1.0),
+
+        x: (-1.0, 1.0 ),
+        ux: (-1.0, 1.0),
+        y: (-1.0, 1.0),
+        uy: (-1.0, 1.0),
+        z: (-1.0, 1.0),
+        uz: (-1.0, 1.0),
+    }
+
+    prob = Problem(
+        obj,
+        obj_grad,
+        eom,
+        state_symbols,
+        num_nodes,
+        interval_value,
+        known_parameter_map=par_map,
+        instance_constraints=instance_constraints,
+        bounds=bounds,
+        time_symbol=t,
+        )
+
+    initial_guess = np.zeros(prob.num_free)
+    start_idx = np.random.randint(0, num_nodes - 1)
+    for i in range(9):
+        initial_guess[start_idx + i*num_nodes] = 10.0
+    with raises(ValueError):
+        prob.bounds_conflict_initial_guess(initial_guess)
+
+    # B: variable time interval
+    h = sym.symbols('h')
+    interval_value = h
+    t0, tf = 0.0, (num_nodes - 1)*h
+
+    def obj(free):
+        Fx = free[6*num_nodes:9*num_nodes]
+        return free[-1]*np.sum(Fx**2)
+
+    def obj_grad(free):
+        grad = np.zeros_like(free)
+        l1 = 6*num_nodes
+        l2 = 9*num_nodes
+        grad[l1: l2] = 2.0*free[l1:l2]*free[-1]
+        grad[-1] = np.sum(free[l1:l2]**2)
+        return grad
+
+    instance_constraints = (
+        x.func(t0),
+        ux.func(t0),
+        x.func(tf) - 1.0,
+        ux.func(tf),
+    )
+    bounds= {
+        a1: (-1.0, 1.0),
+        a2: (-1.0, 1.0),
+        a3: (-1.0, 1.0),
+
+        u1: (-1.0, 1.0),
+        u2: (-1.0, 1.0),
+        u3: (-1.0, 1.0),
+
+        x: (-1.0, 1.0),
+        ux: (-1.0, 1.0),
+        y: (-1.0, 1.0),
+        uy: (-1.0, 1.0),
+        z: (-1.0, 1.0),
+        uz: (-1.0, 1.0),
+        h: (0.0, 0.4 )
+    }
+
+    prob = Problem(
+        obj,
+        obj_grad,
+        eom,
+        state_symbols,
+        num_nodes,
+        interval_value,
+        known_parameter_map=par_map,
+        instance_constraints=instance_constraints,
+        bounds=bounds,
+        time_symbol=t,
+        )
+
+    initial_guess = np.zeros(prob.num_free)
+    start_idx = np.random.randint(0, num_nodes - 1)
+    for i in range(9):
+        initial_guess[start_idx + i*num_nodes] = -10.0
+    initial_guess[-1] = 0.3
+    with raises(ValueError):
+        prob.bounds_conflict_initial_guess(initial_guess)
+
+    # C: respect_bounds=True: initial guess must be within bounds, else a
+    # ValueError is raised.
+    with raises(ValueError):
+        _, _ = prob.solve(initial_guess, respect_bounds=True)
+
+    # D: respect_bounds=False by default. Bounds are ignored.
```

Comment:
you could set max iterations to 1 or 2 to save computation time when solve actually runs.